mcs/class/Mono.Posix/Mono.Unix/UnixEncoding.cs

```diff
@@ -300,17 +300,35 @@ public override int GetByteCount (String s)
 			throw new ArgumentOutOfRangeException ("byteIndex", _("ArgRange_Array"));
 		}
 
+		unsafe {
+			fixed (char *p = s) {
+				fixed (byte *b = bytes) {
+					return GetBytes(p + charIndex, charCount, b + byteIndex, bytes.Length - byteIndex);
```

Comment:
Where is the code that generates the csproj files? Might be worth setting the coding style in them so VSfM stops formatting these badly.